### PR TITLE
Like display system

### PR DESCRIPTION
**Pull request by MadiMalik**
---
name: solution review 
about: A template PR for code review with a checklist
---

The `who_likes_it` function takes a list of names and returns a string describing the number of people who like a post.

# Behavior

## Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/who_likes_it.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Generate a "like display message based on the input list of names
+
+    Args:
+        names (List[str]): A list of names representing people who liked a an item (post)
+
+    Returns:
+        str: A display message that describes how many people liked the post
+
+    Raises:
+        - ValueError: If any name in the list is not a string
```
Comment (mojtabafayyaz) [inline solutions/who_likes_it.py]: **Raises:** 
What do you think about updating the docstring to specify AssertionError instead of ValueError?
Since the code raises AssertionError for invalid inputs, it would be more consistent and clear to reflect this in the docstring. This would ensure the exception type in the documentation matches what is actually raised in the implementation.

**Comment by mojtabafayyaz** [comment]
Dear Madiha,
Your code is clean, well-structured, and easy to follow. The defensive assertions to validate inputs are a nice touch. Your unit tests are thorough, covering edge cases and invalid inputs to ensure the function behaves as expected. The variable names are clear and consistent, making the code easy to understand. Great job!

File: solutions/who_likes_it.py (hunk 1 above)
Comment (MadiMalik) [reply solutions/who_likes_it.py]: Thank you so much for pointing that out! I’ll fix it right away

**Comment by MadiMalik** [comment]
Thank you so much Mojataba, for the valuable feedback!

**Review by mojtabafayyaz — APPROVED** [review]

**Review by mojtabafayyaz — APPROVED** [review]
